File: c/palindrome.c
#include <stdio.h>
#include <string.h>
void main()
{
    int l,c=0;
    char str[20];
    printf("Enter the string : ");
    gets(str);
    l=strlen(str);
    for (int i=0;i<=l/2;i++)
    {
        if (str[i]==str[l-i-1])
            continue;
        else
            c=1;
            break;
    }
    if (c==0)
        printf("\nIt is a palindrome.");
    else   
        printf("\nIt is not a palindrome.");
}

File: Python/Class 11/Programs/palindrome.py
#Check for palindrome
'''s=input('Enter the string : ')
  i=0
  j=len(s)-1
  while i<j:
      if s[i]==s[j]:
          i+=1
          j-=1
      else:
          print('Not a palindrome.')
          break
  else:
      print('Palindrome')'''
s=input('Enter the string :')
if s==s[::-1]:
    print('It is a palindrome.')
else:
    print('Not a palindrome.')
